cmd/upgrade/cluster/cmd.go

```diff
@@ -440,3 +449,39 @@ func run(cmd *cobra.Command, _ []string) {
 
 	reporter.Infof("Upgrade successfully scheduled for cluster '%s'", clusterKey)
 }
+
+func checkAndAckMissingAgreements(ocmClient *ocm.Client, cluster *cmv1.Cluster, upgradePolicy *cmv1.UpgradePolicy,
+	reporter *rprtr.Object, clusterKey string) error {
+	// check if the cluster upgrade requires gate agreements
+	gates, err := ocmClient.GetMissingGateAgreements(cluster.ID(), upgradePolicy)
+	if err != nil {
+		return errors.Errorf("Failed to check for missing gate agreements upgrade for "+
+			"cluster '%s': %v", clusterKey, err)
+	}
+	isWarningDisplayed := false
+	for _, gate := range gates {
+		if !gate.STSOnly() {
+			if isWarningDisplayed {
+				reporter.Warnf("Missing required acknowledgements to schedule upgrade. Read the below " +
+					"description and acknowledge to proceed with upgrade \n")
+				isWarningDisplayed = true
+			}
+			// Print short cluster description:
+			str := fmt.Sprintf("\t"+
+				"Description: %s\n"+
+				"\tURL:         %s\n",
+				ocm.WordWrap(gate.Description(), 160), gate.DocumentationURL())
+			reporter.Infof("%s\n", str)
```

Comment:
We have `interactive.PrintHelp`, which fits in the same output style and model of the interactive prompts. See the various `rosa create idp` implementations for examples.